exchanges/exchange.go

```diff
@@ -262,35 +258,54 @@ func (e *Base) SetCurrencyPairFormat() {
 
 	assetTypes := e.GetAssetTypes()
 	for x := range assetTypes {
-		if e.Config.CurrencyPairs.Get(assetTypes[x]) == nil {
-			r := e.CurrencyPairs.Get(assetTypes[x])
-			if r == nil {
+		if _, err := e.Config.CurrencyPairs.Get(assetTypes[x]); err != nil {
+			ps, err := e.CurrencyPairs.Get(assetTypes[x])
+			if err != nil {
 				continue
 			}
-			e.Config.CurrencyPairs.Store(assetTypes[x], *e.CurrencyPairs.Get(assetTypes[x]))
+			e.Config.CurrencyPairs.Store(assetTypes[x], *ps)
 		}
 	}
 }
 
 // SetConfigPairs sets the exchanges currency pairs to the pairs set in the config
-func (e *Base) SetConfigPairs() {
-	assetTypes := e.GetAssetTypes()
+func (e *Base) SetConfigPairs() error {
+	assetTypes := e.Config.CurrencyPairs.GetAssetTypes()
+	exchageAssets := e.CurrencyPairs.GetAssetTypes()
 	for x := range assetTypes {
-		cfgPS := e.Config.CurrencyPairs.Get(assetTypes[x])
-		if cfgPS == nil {
-			continue
+		if !exchageAssets.Contains(assetTypes[x]) {
+			log.Warnf(log.ExchangeSys,
+				"%s exchange asset type %s unsupported, please manually remove from configuration or add it to the exchange pairs management systen in %s_wrapper.go",
```

Comment:
systen -> system 

But also not a fan of directly referencing source files like this in a log.Warn

Reply:
Yeah dropped the reference completely.